### PR TITLE
Implement effect of saloon card #33

### PR DESCRIPTION
**Pull request by IvanovAndrew**

### CODE REVIEW

File: Bang.Tests/SaloonSpecification.cs [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+
+            // act
+            actor.PlayCard(saloonCard);
+            
+            // Assert
+            gameplay.GetTopCardFromDiscarded().Should().Be(saloonCard);
+        }
+        
+        [Fact]
+        public void Player_discards_gatling_card_after_it_played()
```
Comment (bashnia007) [inline Bang.Tests/SaloonSpecification.cs:31]: Copypaste victim :) Not gatling but saloon

File: Bang.Tests/SaloonSpecification.cs [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+
+            // act
+            actor.PlayCard(saloonCard);
+            
+            // assert
+            otherPlayer.LifePoints.Should().Be(health);
+        }
+        
+        [Fact]
+        public void Saloon_regains_one_life_point_to_current_player()
```
Comment (bashnia007) [inline Bang.Tests/SaloonSpecification.cs:62]: What do you think about joining Saloon_regains_one_life_point_to_all_alive_players and Saloon_regains_one_life_point_to_current_player ?

File: Bang.Tests/SaloonSpecification.cs [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+
+            // act
+            actor.PlayCard(saloonCard);
+
+            // Assert
+            actor.Hand.Should().NotContain(saloonCard);
+        }
+
+        [Fact]
+        public void Saloon_regains_one_life_point_to_all_alive_players()
```
Comment (bashnia007) [inline Bang.Tests/SaloonSpecification.cs:44]: Name is missleading. Here you check all alive players except current one

File: Bang.Tests/SaloonSpecification.cs (hunk 1 above)
Comment (IvanovAndrew) [reply Bang.Tests/SaloonSpecification.cs]: Renamed :-)

File: Bang.Tests/SaloonSpecification.cs (hunk 2 above)
Comment (IvanovAndrew) [reply Bang.Tests/SaloonSpecification.cs]: Joined into one method

**Pull request merged**